packages/odata-service-writer/src/data/annotations.ts

```diff
@@ -0,0 +1,97 @@
+import { XMLParser, XMLValidator } from 'fast-xml-parser';
+import { t } from '../i18n';
+import { NamespaceAlias, OdataService } from '../types';
+
+/**
+ * Updates the passed odata service with the namespaces parsed from the specified metadata and annotations.
+ * 
+ * @param {Partial<OdataService>} service - an odata service where at least metadata and annotations properties are defined
+ * @returns A reference to the namspaces array
+ */
+export function getAnnotationNamespaces(service: Partial<OdataService>): NamespaceAlias[] {
+    // enhance service with annotations namespaces
+    const schemaNamespaces = service.metadata ? getNamespaces(service.metadata) : [];
+
+    return schemaNamespaces.map((schema: NamespaceAlias) => {
+        // Check if alias exists in backend annotation file, if so use it
+        if (service.annotations) {
+            const annotationAlias =
+                service.annotations.xml && schema.namespace
+                    ? getAliasFromAnnotation(service.annotations.xml, schema.namespace)
+                    : '';
+            if (annotationAlias) {
+                schema.alias = annotationAlias;
+            }
+        }
+        return schema;
+    });
+}
+
+/**
+ * Convert specified xml string to JSON.
+ *
+ * @param xml - the schema to parse
+ * @returns parsed object representation of passed XML
+ */
+function xmlToJson(xml: string): any | void {
+    if (XMLValidator.validate(xml) !== true) {
+        throw new Error(t('error.invalidXML'));
+    }
+    const options = {
+        attributeNamePrefix: '',
+        ignoreAttributes: false,
+        ignoreNameSpace: true,
+        parseAttributeValue: true
+    };
+    const parser = new XMLParser(options);
+    return parser.parse(xml);
```

Comment:
Need a try catch here on the call to 'parse'?

Reply:
I dont think so since we validate the xml here https://github.com/SAP/open-ux-tools/pull/298/files/3e6e0ce7283018ce3d19fd73d0bc3f7169f28155#diff-f507de448682b891118eb30af7f854e31a976fa9ccdcfc1818dea12ef9a296eaR37